inference-engine/thirdparty/clDNN/src/graph_optimizer/concat_input_order.cpp

```diff
@@ -0,0 +1,223 @@
+// Copyright (c) 2020 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+///////////////////////////////////////////////////////////////////////////////////////////////////
+
+#include "pass_manager.h"
+#include "pooling_inst.h"
+#include "convolution_inst.h"
+#include "fully_connected_inst.h"
+#include "data_inst.h"
+#include "memory_impl.h"
+#include "program_impl.h"
+
+#include <vector>
+#include <tuple>
+
+using namespace cldnn;
+
+namespace {
+
+using shuffle_range = std::pair<int32_t, int32_t>;
+
+bool can_shuffle_features(program_node& node) {
+    if (node.is_type<convolution>()) {
+        auto& conv_node = node.as<convolution>();
+        auto& wei_node = conv_node.weights();
+
+        return conv_node.get_groups() == 1 && conv_node.get_split() == 1 &&
+            conv_node.get_deformable_groups() == 1 && !conv_node.get_transposed() &&
+            !conv_node.activations_zero_points_term() &&
```

Comment:
What about weights zero points?

Reply:
Aren't they always per output feature? So changing input features order shouldn't affect that.

Reply:
OK, you're right